### ISSUE TITLE
unable to install perl 5.36.1 on AIX: op/sprintf2.t test script is failing

### ISSUE THREAD

**Issue by XSven** [Perl/perl5#21251]
**Module**
perl 5.36.1

**Description**
I am trying to install perl 5.36.1 using perlbrew (0.97 most recent) on an AIX system (7300-01-02-2320).

```
# Failed test 1704 - sprintf( "%.54g", 0.3 ) renders correctly at op/sprintf2.t line 1197
#      got "0.2999999999999999888977697537484345958"
# expected eq "0.299999999999999988897769753748434595763683319091796875"
op/sprintf2.t ........................................................
Failed 1/1704 subtests
        (less 30 skipped subtests: 1673 okay)
```

**Expected behavior**
test should pass

**perlbrew configure flags**
```
PERLBREW_CONFIGURE_FLAGS=-de -Duse64bitall -Dusethreads -Duselargefiles -Dcc='gcc -maix64 -D_XOPEN_SOURCE=700' -Doptimize=-O2 -Dlibs='-lpthread -lbind -lnsl -lgdbm -ldbm -ldl -lld -lm -lcrypt -lpthreads -lc' -Dldflags='-Wl,-brtl,-bexpall,-bmaxdata:0x80000000,-blibpath:/usr/local/lib:/opt/freeware/lib:/usr/lib:/lib'
```

**Comment by demerphq** [comment]
I believe that this is fixed in 5.38. Could you check? I believe it is a test failure. Some platforms do not support as many sprintf digits as others.

**Comment by XSven** [comment]
Yes sorry it is, but in 5.38 I have another failing test [21249](https://github.com/Perl/perl5/issues/21249).
